src/components/layouts/FilterBar.tsx

```diff
@@ -0,0 +1,31 @@
+'use client'
+
+import { Box, Stack, Typography } from '@mui/material'
+import { AppMargin, SizeofAppMargin } from '@/hoc/AppMargin'
+import { useEffect, useState } from 'react'
+import SearchBar from '../searchBar/SearchBar'
+import store from '@/redux/store'
+import { ClientSideStateUpdate } from '@/hoc/ClientSideStateUpdate'
+import { setFilteredTasks } from '@/redux/features/taskBoardSlice'
+
+export const FilterBar = ({}: {}) => {
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    store.dispatch(setFilteredTasks(searchText))
+  }, [searchText])
```

Comment:
I think this `useEffect` is not needed here. Why don't we directly dispatch the reducer `setFilteredTasks` from the `getSearchKeyword` event? 
```
<SearchBar value={searchText} getSearchKeyword={(keyword) => {
            setSearchText(keyword)
            store.dispatch(setFilteredTasks(searchText))
}} />
```
@arpandhakal